habitat_sim/agent/agent.py

```diff
@@ -226,6 +229,9 @@ def set_state(self, state: AgentState, reset_sensors: bool = True):
             )
             s.node.rotation = quat_to_magnum(state.rotation.inverse() * v.rotation)
 
+        if is_initial:
+            self.initial_state = state
```

Comment:
Maybe we should set initial_state from agent.reconfigure?

Reply:
This makes sense to me.  I like the separation of agent configuration from agent state (that keeps the agent configuration from being specific to an environment).